<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <meta name="author" content="Variant annotation" />
  <title>NGS data analysis course</title>
  <style type="text/css">code{white-space: pre;}</style>
  <link rel="stylesheet" href="../../../Commons/css_template_for_examples.css" type="text/css" />
</head>
<body>
<div id="header">
<h1 class="title"><a href="http://ngscourse.github.io/">NGS data analysis course</a></h1>
<h2 class="author"><strong>Variant annotation</strong></h2>
<h3 class="date"><em>(updated 08-06-2014)</em></h3>
</div>
<!-- COMMON LINKS HERE -->

<h1 id="variant-annotation-with-annovar">Variant annotation with Annovar</h1>
<p>Copy the necessary data in your working directory:</p>
<pre><code>mkdir -p /home/participant/cambridge_mda14/
cp -r /home/participant/Desktop/Open_Share/annotation /home/participant/cambridge_mda14/
cd /home/participant/cambridge_mda14/annotation/annovar</code></pre>
<h2 id="vcf-to-annovar-format">1. VCF to Annovar format</h2>
<pre><code>perl convert2annovar.pl -format vcf4 example/example1.vcf &gt; example/example1.annovar</code></pre>
<p>The above command takes example1.vcf as input file, and generate the example1.annovar as output file. The 3 extra columns are zygosity status, genotype quality and read depth.</p>
<p>If you read the screen message carefully, it tells that only 1 out of 3 samples have been processed in this VCF file. <strong>By default, only the first sample in VCF file will be written to output file</strong>. The input contains five loci, but two of them do not have variation for the first sample, and that is why the output contains only 3 variants.</p>
<h2 id="download-gene-annotation-database-for-hg18-build-and-save-to-humandb-directory">1. Download gene annotation database (for hg18 build) and save to humandb/ directory</h2>
<pre><code>perl annotate_variation.pl -downdb refGene humandb/

perl annotate_variation.pl -downdb snp128 humandb/

perl annotate_variation.pl -downdb 1000g2012apr humandb/</code></pre>
<p>Other possible downloads for hg19 (more can be found at http://www.openbioinformatics.org/annovar/annovar_download.html):</p>
<pre><code>perl annotate_variation.pl -buildver hg19 -downdb refGene humandb/
perl annotate_variation.pl -buildver hg19 -downdb phastConsElements46way humandb/
perl annotate_variation.pl -buildver hg19 -downdb genomicSuperDups humandb/
perl annotate_variation.pl -buildver hg19 -downdb 1000g2012apr humandb/
perl annotate_variation.pl -buildver hg19 -downdb snp135 humandb/ 
perl annotate_variation.pl -buildver hg19 -downdb ljb2_all humandb/
perl annotate_variation.pl -buildver hg19 -downdb esp6500si_all humandb/</code></pre>
<h2 id="gene-based-annotation-of-variants-in-the-varlist-file-by-default-geneanno-is-on">2. Gene-based annotation of variants in the varlist file (by default â€“geneanno is ON)</h2>
<pre><code>mkdir results
perl annotate_variation.pl --geneanno example/ex1.human humandb/ --outfile results/0-geneanno
      </code></pre>
<h2 id="region-based-annotate-variants">3. Region-based annotate variants</h2>
<pre><code>perl annotate_variation.pl -regionanno -dbtype cytoBand example/ex1.human humandb/ --outfile results/1-regionanno</code></pre>
<h2 id="filter-rare-or-unreported-variants-in-1000gdbsnp-or-predicted-deleterious-variants">4. Filter rare or unreported variants (in 1000G/dbSNP) or predicted deleterious variants</h2>
<pre><code>perl annotate_variation.pl -filter -dbtype 1000g2012apr_all -maf 0.01 example/ex1.human humandb/ --outfile results/2-filter

perl annotate_variation.pl -filter -dbtype snp128 example/ex1.human humandb/ --outfile results/2-filter</code></pre>
</body>
</html>
